ui/app/AppLayouts/Profile/panels/ProfileShowcasePanel.qml

```diff
@@ -0,0 +1,351 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import Qt.labs.settings 1.0
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Popups 0.1
+
+import utils 1.0
+
+import AppLayouts.Profile.controls 1.0
+
+Control {
+    id: root
+
+    property var baseModel
+
+    readonly property alias settings: settings
+    readonly property alias showcaseModel: showcaseModel
+
+    // to override
+    property string settingsKey
+    property string keyRole
+    property var roleNames: []
+    property var filterFunc: (modelData) => true
+    property string hiddenPlaceholderBanner
+    property string showcasePlaceholderBanner
+    property Component draggableDelegateComponent
+    property Component showcaseDraggableDelegateComponent
+
+    background: null
+
+    component VisibilityDropArea: AbstractButton {
+        id: visibilityDropAreaLocal
+
+        property int showcaseVisibility: Constants.ShowcaseVisibility.NoOne
+        readonly property alias containsDrag: dropArea.containsDrag
+
+        padding: Style.current.halfPadding
+        spacing: padding/2
+
+        icon.color: Theme.palette.primaryColor1
+
+        background: ShapeRectangle {
+            path.strokeColor: dropArea.containsDrag ? "transparent" : Theme.palette.directColor7
+            path.fillColor: dropArea.containsDrag ? Theme.palette.white : "transparent"
+
+            DropArea {
+                id: dropArea
+                anchors.fill: parent
+                keys: ["x-status-draggable-showcase-item-hidden"]
+                onEntered: function(drag) {
+                    drag.accept()
+                }
+
+                onDropped: function(drop) {
+                    var showcaseObj = drop.source.showcaseObj
+
+                    var tmpObj = Object()
```

Comment:
It feels weird to declare sth as `const` and then modify it a few lines below

Reply:
in js const refers to references, not object's itself, so it would be fully valid to mark it as `const`. That reference points the same object here from beginning to the end of the scope.